.github/workflows/ci-tls.yml

```diff
@@ -0,0 +1,76 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This workflow tests the TLS configuration of the smithy-rs client
+
+env:
+  rust_version: 1.68.2
+
+name: Verify client TLS configuration
+on:
+  pull_request:
+
+jobs:
+  verify-tls-config:
+    name: Verify TLS configuration
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install packages
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install gcc make python3-pip nginx git ruby openjdk-17-jre pkg-config libssl-dev faketime
+          pip3 install certbuilder crlbuilder
+      - name: Stop nginx
+        run: sudo systemctl stop nginx
+      - name: Checkout smithy-rs
+        uses: actions/checkout@v3
+        with:
+          path: ./smithy-rs
+      - name: Checkout trytls
+        uses: actions/checkout@v3
+        with:
+          repository: ouspg/trytls
+          path: ./trytls
+      - name: Checkout badtls
+        uses: actions/checkout@v3
+        with:
+          repository: wbond/badtls.io
+          path: ./badtls.io
+      - name: Checkout badssl
+        uses: actions/checkout@v3
+        with:
+          repository: chromium/badssl.com
+          path: ./badssl.com
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.rust_version }}
+      - name: Build badssl.com
+        shell: bash
+        working-directory: badssl.com
+        env:
+          DOCKER_BUILDKIT: 1
```

Comment:
is this used? is this step actually using Docker?

Reply:
Yes, it's `make serve` in the script